processor/manager.go

```diff
@@ -32,25 +34,29 @@ func (proc *LifecycleManager) Run(ctx context.Context) error {
 // StartNew starts a processor, this is not a blocking call.
 //If the processor is not completely started and the data started coming then also it will not be problematic as we
 //are assuming that the DBs will be up.
-func (proc *LifecycleManager) StartNew() {
-	proc.HandleT.Setup(proc.backendConfig, proc.gatewayDB, proc.routerDB, proc.batchRouterDB,
-		proc.errDB, proc.clearDB, proc.reporting, proc.multitenantStats)
+func (proc *LifecycleManager) Start() {
+	if proc.Transformer != nil {
```

Comment:
this appears to be a late stage for performing configuration changes. Why is it necessary and why can't we do this earlier, when the `LifecycleManager` is being created?

Reply:
This pattern of lazy inti is quite common in Go. There is no `New` function for `LifecycleManager`

Reply:
Instead of having an exported struct field `Transformer` which ultimately acts as an **indirect setter** for the private field `HandleT.transformer` wouldn't it be simpler and more straightforward to omit this field from the struct, remove any special handling from the `Start` method and add the missing setter method?

```go
func (proc *LifecycleManager) SetTransformer(t transformer.Transformer) {
    proc.HandleT.transformer = proc.Transformer
}
```